shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/HardFork.hs

```diff
@@ -0,0 +1,171 @@
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Shelley.Spec.Ledger.HardFork
+  ( mkInitialShelleyLedgerView,
+    translateToShelleyLedgerState,
+      -- * Exported for testing purposes
+    translateTxIdByronToShelley,
+    translateCompactTxOutByronToShelley
+  ) where
+
+import qualified Cardano.Chain.Block as Byron
+import qualified Cardano.Chain.Common as Byron
+import qualified Cardano.Chain.UTxO as Byron
+import qualified Cardano.Crypto.Hash.Class as Byron
+import qualified Cardano.Crypto.Hashing as Hashing
+import Control.Monad.Reader (runReader)
+import qualified Data.Map.Strict as Map
+import Data.Maybe (fromMaybe)
+import qualified Data.ByteString as Strict
+import GHC.Stack (HasCallStack)
+import Shelley.Spec.Ledger.API
+import Shelley.Spec.Ledger.BaseTypes
+import Shelley.Spec.Ledger.Crypto
+import Shelley.Spec.Ledger.EpochBoundary
+import Shelley.Spec.Ledger.Genesis
+import Shelley.Spec.Ledger.LedgerState
+import Shelley.Spec.Ledger.OverlaySchedule
+import Shelley.Spec.Ledger.Rewards
+import Shelley.Spec.Ledger.Slot
+import Shelley.Spec.Ledger.UTxO
+
+-- | We use the same hashing algorithm so we can unwrap and rewrap the bytes.
+-- We don't care about the type that is hashed, which will differ going from
+-- Byron to Shelley, we just use the hashes as IDs.
+translateTxIdByronToShelley :: Crypto sc => Byron.TxId -> TxId sc
+translateTxIdByronToShelley =
+    TxId . hashFromBytesE . Hashing.hashToBytes
```

Comment:
Actually, we can improve this (possibly in a separate commit): both types are `ShortByteString`s under the hood, so we should avoid converting via a `ByteString`.